op-e2e/actions/upgrades/utils/assertions.go

```diff
@@ -0,0 +1,32 @@
+package utils
+
+import (
+	"math/big"
+
+	actionhelpers "github.com/ethereum-optimism/optimism/op-e2e/actions/helpers"
+	"github.com/ethereum-optimism/optimism/op-e2e/actions/interop/dsl"
+	"github.com/ethereum-optimism/optimism/op-service/eth"
+	"github.com/ethereum/go-ethereum"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/lmittmann/w3"
+	"github.com/stretchr/testify/require"
+)
+
+var ProxyImplGetterFunc = w3.MustNewFunc(`implementation()`, `address`)
+
+func RequireContractDeployedAndProxyUpdated(t actionhelpers.Testing, chain *dsl.Chain, implAddr common.Address, proxyAddress common.Address, activationBlockID eth.BlockID) {
```

Comment:
This function still only requires the block number, so I'd revert the change to pass a `BlockID` back to a `*big.Int` block number. Let's do this in the follow-up.

Reply:
TODO in follow-up